src/Microgrammar.ts

```diff
@@ -102,6 +105,27 @@ export class Microgrammar<T> implements Term {
         return found.length > 0 ? found[0] : null;
     }
 
+    /**
+     * Return a match if it explains the whole of the input.
+     * This style of usage is more like a traditional parser,
+     * building an AST for a while file.
+     * @param input
+     * @return {PatternMatch&T}
+     */
+    // TODO we should allow InputStream also, although this makes
+    // verifying that we got to the end harder
+    public exactMatch(input: string): PatternMatch & T {
+        const match = this.firstMatch(input);
```

Comment:
this is an opportunity for optimization later, eh? Asking for an exact match is more information that we could use to go faster. But that can come later. Adding it to the API lets people give us that info so that later we can make it go faster.
This would be a good place to put a DismatchReport sort of return